### PR TITLE
Azure AD / Entra ID custom authentication extension

### PR DESCRIPTION
**Pull request by AldoSGNL**
Hi Marc! Please review the refactored code and readme for the Azure AD custom authentication extension. Thanks!

### CODE REVIEW

File: azuread/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Azure AD / Entra ID App Function HTTP Trigger Example
+This C# script example demonstrates how to call the SGNL Access Service from an Azure App Function HTTP Trigger. This app function and trigger is then configured as part of an custom authentication extension.
```
Comment (marc-sgnl) [inline azuread/README.md:2]:
```suggestion
This example uses Custom Authentication Extension in Entra Id (formerly Azure AD) to make a request for an access decision during sign-in to Applications/Services configured in Microsoft Entra. The sample can be run in Azure App Function, is written in C#, and can be modified to support additional configuration and/or metadata that may be configured for your applications.
```

File: azuread/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Prerequisites
+
+1. You have a SGNL environment up and running.
+ 
+2. You have configured connectors to ingest data and have defined the appropriate relationship mappings.
+ 
+3. You have created policy snippets and policy for evaluation.
+ 
+4. You have created an protected application to test. You can test with this [example](https://jwt.ms/) provided by Microsoft.
```
Comment (marc-sgnl) [inline azuread/README.md:13]:
```suggestion
4. You have created an Entra ID Application to test. You can test with this [example](https://jwt.ms/) provided by Microsoft.
```

File: azuread/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ 
+5. You will need Git to clone the example repository. You can follow the steps to install Git [here](https://github.com/git-guides/install-git).
+
+
+See our [Help Guides](https://support.sgnl.ai) for steps on configuring data sources and policies.
+
+
+## Steps For Configuring the custom authentication extension.
```
Comment (marc-sgnl) [inline azuread/README.md:21]: nitpick: your capitalization on your headings is inconsistent - I'm not partial to which way you go, but could you update them all to either sentence or title case?

File: azuread/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+1. Clone the SGNL examples repo using Git from https://github.com/SGNL-ai/examples.git.
+2. Login to your Azure portal.
```
Comment (marc-sgnl) [inline azuread/README.md:25]:
```suggestion
2. Login to the Azure portal.
```

File: azuread/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+3. Follow the steps found [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/custom-extension-get-started?tabs=azure-portal%2Chttp).
```
Comment (marc-sgnl) [inline azuread/README.md:26]:
```suggestion
3. Follow the steps found [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/custom-extension-get-started?tabs=azure-portal%2Chttp) to configure a custom extension.
```

File: azuread/README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+4. Replace the default C# script in the example in step 3, with the run.csx C# code from the SGNL examples repository.
```
Comment (marc-sgnl) [inline azuread/README.md:27]: I don't know that I'd call this a script, it feels like legitimate code to me. Thoughts?

File: azuread/README.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+
+## Steps For Testing The Example
+
+1. Once your configuration is complete, you can build your login URL for initiating the Azure AD IDP authentication flow. See the example URL below:
```
Comment (marc-sgnl) [inline azuread/README.md:32]: I'd suggest just using the myapps portal - this seems more complex than necessary.

File: azuread/run.csx [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        public string DateOfBirth { get; set; }
+        public string ApiVersion { get; set; }
+        public List<string> CustomRoles { get; set; }
+        public Claims()
+        {
+        CustomRoles = new List<string>();
+        }
+    }
+
+    // Class definition for the query sent to the SGNL access service api.
```
Comment (marc-sgnl) [inline azuread/run.csx:181]:
```suggestion
    // Class definition for the query sent to the SGNL Access Service.
```

File: azuread/run.csx [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+    public class Query {
+        public string action {get;set;} = "";
+        public string assetId {get;set;} = "";
+    }
+
+    // Class definition for the principal sent to the SGNL access service api.
```
Comment (marc-sgnl) [inline azuread/run.csx:188]:
```suggestion
    // Class definition for the principal sent to the SGNL Access Service.
```

File: azuread/run.csx [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+
+    public class AzurePrincipal {
+        public string id {get;set;}="";
+    }
+
+
+    // Class definition for the entire request JSON sent to the SGNL access service api.
```
Comment (marc-sgnl) [inline azuread/run.csx:195]:
```suggestion
    // Class definition for the entire request JSON sent to the SGNL Access Service.
```

File: azuread/README.md (hunk 6 above)
Comment (AldoSGNL) [reply azuread/README.md:27]: I think MS categorizes a csx file as a script: https://learn.microsoft.com/en-us/archive/msdn-magazine/2016/january/essential-net-csharp-scripting

File: azuread/README.md (hunk 7 above)
Comment (AldoSGNL) [reply azuread/README.md:32]: It was actually pretty easy to implement.

File: azuread/README.md (hunk 3 above)
Comment (AldoSGNL) [reply azuread/README.md:21]: Got it! I will update to title case.

File: azuread/README.md (hunk 6 above)
Comment (AldoSGNL) [reply azuread/README.md:27]: @marc-sgnl All set, mate. I gave the reader two options to test the custom authentication extension. Thanks!

**Review by AldoSGNL — COMMENTED** [review]
@marc-sgnl All set mate. Please review the updated Readme. I added two options to test the custom authentication extension.

**Review by marc-sgnl — DISMISSED** [review]

**Review by marc-sgnl — APPROVED** [review]

**Pull request merged**